test/parallel/test-inspector-connect-main-thread.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+const common = require('../common');
+common.skipIfInspectorDisabled();
+const assert = require('assert');
+const { Session } = require('inspector');
+const path = require('path');
+const { pathToFileURL } = require('url');
+const { isMainThread, parentPort, Worker } = require('worker_threads');
+
+function toDebug() {
+  let a = 1;
+  a = a + 1;
+  return a * 200;
+}
+
+const messagesSent = [];
+
+// This ensures that worker does not terminate while waiting for
+// inspector response
+function wrapCallbackToKeepThreadAlive(callback) {
```

Comment:
Do all clients of the new API will need something like this?

Reply:
This will be fixed separately.